<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>splicejac.plot.grn_plots &mdash; spliceJAC 0.1 documentation</title>
      <link rel="stylesheet" href="../../../../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../../../../_static/css/theme.css" type="text/css" />
      <link rel="stylesheet" href="../../../../_static/graphviz.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../../../../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../../../../" id="documentation_options" src="../../../../_static/documentation_options.js"></script>
        <script src="../../../../_static/jquery.js"></script>
        <script src="../../../../_static/underscore.js"></script>
        <script src="../../../../_static/doctools.js"></script>
        <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    <script src="../../../../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../../../../genindex.html" />
    <link rel="search" title="Search" href="../../../../search.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="../../../../index.html" class="icon icon-home"> spliceJAC
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../../usage.html">Usage</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../notebooks/GRN%20Inference.html">GRN Inference</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../notebooks/Transitions.html">Transitions</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../index.html">API Reference</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../../../index.html">spliceJAC</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../../../../index.html" class="icon icon-home"></a> &raquo;</li>
      <li><code class="xref py py-mod docutils literal notranslate"><span class="pre">splicejac.plot.grn_plots</span></code></li>
      <li class="wy-breadcrumbs-aside">
            <a href="../../../../_sources/autoapi/splicejac/plot/grn_plots/index.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  
<style>
/* CSS overrides for sphinx_rtd_theme */

/* 24px margin */
.nbinput.nblast.container,
.nboutput.nblast.container {
    margin-bottom: 19px;  /* padding has already 5px */
}

/* ... except between code cells! */
.nblast.container + .nbinput.container {
    margin-top: -19px;
}

.admonition > p:before {
    margin-right: 4px;  /* make room for the exclamation icon */
}

/* Fix math alignment, see https://github.com/rtfd/sphinx_rtd_theme/pull/686 */
.math {
    text-align: unset;
}
</style>
<section id="module-splicejac.plot.grn_plots">
<span id="splicejac-plot-grn-plots"></span><h1><a class="reference internal" href="#module-splicejac.plot.grn_plots" title="splicejac.plot.grn_plots"><code class="xref py py-mod docutils literal notranslate"><span class="pre">splicejac.plot.grn_plots</span></code></a><a class="headerlink" href="#module-splicejac.plot.grn_plots" title="Permalink to this headline"></a></h1>
<p>plotting resources for GRN visualization</p>
<section id="module-contents">
<h2>Module Contents<a class="headerlink" href="#module-contents" title="Permalink to this headline"></a></h2>
<section id="functions">
<h3>Functions<a class="headerlink" href="#functions" title="Permalink to this headline"></a></h3>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><a class="reference internal" href="#splicejac.plot.grn_plots.plot_grn" title="splicejac.plot.grn_plots.plot_grn"><code class="xref py py-obj docutils literal notranslate"><span class="pre">plot_grn</span></code></a>(G[, node_size, edge_width, font_size, adata, ...])</p></td>
<td><p>Plot the GRN with positive and negative interactions</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#splicejac.plot.grn_plots.visualize_network" title="splicejac.plot.grn_plots.visualize_network"><code class="xref py py-obj docutils literal notranslate"><span class="pre">visualize_network</span></code></a>(adata, cluster_name[, genes, cc_id, ...])</p></td>
<td><p>Plot the gene regulatory network of a cluster</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#splicejac.plot.grn_plots.diff_network" title="splicejac.plot.grn_plots.diff_network"><code class="xref py py-obj docutils literal notranslate"><span class="pre">diff_network</span></code></a>(adata, cluster1, cluster2[, genes, ...])</p></td>
<td><p>Plot the differential network between two cell states</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#splicejac.plot.grn_plots.diff_interactions" title="splicejac.plot.grn_plots.diff_interactions"><code class="xref py py-obj docutils literal notranslate"><span class="pre">diff_interactions</span></code></a>(adata, cluster1, cluster2[, ...])</p></td>
<td><p>Plot the top differential interactions between two cell states</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#splicejac.plot.grn_plots.conserved_grn" title="splicejac.plot.grn_plots.conserved_grn"><code class="xref py py-obj docutils literal notranslate"><span class="pre">conserved_grn</span></code></a>(adata, cluster1, cluster2[, genes, ...])</p></td>
<td><p>Plot the conserved network between two cell states</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#splicejac.plot.grn_plots.top_conserved_int" title="splicejac.plot.grn_plots.top_conserved_int"><code class="xref py py-obj docutils literal notranslate"><span class="pre">top_conserved_int</span></code></a>(adata, cluster1, cluster2[, ...])</p></td>
<td><p>Plot the top conserved interactions between two cell states</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#splicejac.plot.grn_plots.robust_mean" title="splicejac.plot.grn_plots.robust_mean"><code class="xref py py-obj docutils literal notranslate"><span class="pre">robust_mean</span></code></a>(X)</p></td>
<td><p>Calculate gene expression mean based on quantile selection</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#splicejac.plot.grn_plots.NormalizeData" title="splicejac.plot.grn_plots.NormalizeData"><code class="xref py py-obj docutils literal notranslate"><span class="pre">NormalizeData</span></code></a>(data)</p></td>
<td><p>normalize the gene expression data</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#splicejac.plot.grn_plots.circle_pos" title="splicejac.plot.grn_plots.circle_pos"><code class="xref py py-obj docutils literal notranslate"><span class="pre">circle_pos</span></code></a>(G)</p></td>
<td><p>Compute coordinates along a unit circle for GRN node positions</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#splicejac.plot.grn_plots.subset_jacobian" title="splicejac.plot.grn_plots.subset_jacobian"><code class="xref py py-obj docutils literal notranslate"><span class="pre">subset_jacobian</span></code></a>(J, genes, genelist)</p></td>
<td><p>select gene-gene interactions for genes in the genelist</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#splicejac.plot.grn_plots.core_GRN" title="splicejac.plot.grn_plots.core_GRN"><code class="xref py py-obj docutils literal notranslate"><span class="pre">core_GRN</span></code></a>(adata, cluster1, cluster2[, type_color, ...])</p></td>
<td><p>Plot a reduced GRN including the top DEG of the starting cluster and the top transition genes</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#splicejac.plot.grn_plots.bif_GRN" title="splicejac.plot.grn_plots.bif_GRN"><code class="xref py py-obj docutils literal notranslate"><span class="pre">bif_GRN</span></code></a>(adata, start, end[, pos_edge_color, ...])</p></td>
<td><p>Plot the reduced GRN of transition genes involved in different cell state transitions</p></td>
</tr>
</tbody>
</table>
<dl class="py function">
<dt class="sig sig-object py" id="splicejac.plot.grn_plots.plot_grn">
<span class="sig-prename descclassname"><span class="pre">splicejac.plot.grn_plots.</span></span><span class="sig-name descname"><span class="pre">plot_grn</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">node_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">200</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">edge_width</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">font_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">8</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">adata</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">node_color</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'centrality'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cluster_name</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pos_style</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'spring'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">base_node_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">300</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">diff_node_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">600</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pos_edge_color</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'b'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">neg_edge_color</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'r'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">arrowsize</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">arrow_alpha</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.75</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">conn_style</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'straight'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">colorbar</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">fontweight</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'normal'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#splicejac.plot.grn_plots.plot_grn" title="Permalink to this definition"></a></dt>
<dd><p>Plot the GRN with positive and negative interactions</p>
<dl class="simple">
<dt>G: <cite>networkx graph object</cite></dt><dd><p>network to plot</p>
</dd>
<dt>node_size: <cite>int</cite> or <cite>str</cite> (default: 200)</dt><dd><p>size of nodes. If node_size=’expression’, the node size is scaled based on gene expression.
Otherwise, node_size can be a fixed integer</p>
</dd>
<dt>edge_width: <cite>float</cite> (default: 1)</dt><dd><p>width of connection arrows</p>
</dd>
<dt>fontsize: <cite>int</cite> (default: 10)</dt><dd><p>fontsize for node labels.</p>
</dd>
<dt>adata: <cite>~anndata.AnnData</cite> or <cite>None</cite> (default: <cite>None</cite>)</dt><dd><p>count matrix, must be provided if node_size == ‘expression’</p>
</dd>
<dt>node_color: <cite>str</cite> (default: ‘centrality’)</dt><dd><p>color of nodes. If node_color=’centrality’, the color is based on the node centrality.
Otherwise, a matplotlib color name can be provided. A full list of accepted named colors can be found at:
<a class="reference external" href="https://matplotlib.org/stable/gallery/color/named_colors.html">https://matplotlib.org/stable/gallery/color/named_colors.html</a></p>
</dd>
<dt>cluster_name: <cite>str</cite> or <cite>None</cite> (default: <cite>None</cite>)</dt><dd><p>name of considered cluster (must be provided if node_size=’expression’)</p>
</dd>
<dt>pos_style: <cite>str</cite> (default: ‘spring’)</dt><dd><p>position of nodes. The options are ‘spring’ and ‘circle’. ‘spring’ uses the networkx spring position function,
whereas ‘circle’ arranges the nodes in a circle.</p>
</dd>
<dt>base_node_size: <cite>float</cite> (default: 300)</dt><dd><p>Minimum node size (used if node_size=’expression’)</p>
</dd>
<dt>diff_node_size: <cite>float</cite> (default: 600)</dt><dd><p>difference between minimum and maximal node size (used if node_size=’expression’)</p>
</dd>
<dt>pos_edge_color: <cite>str</cite> (default: ‘b’)</dt><dd><p>color for positive regulation arrow. A full list of accepted named colors can be found at:
<a class="reference external" href="https://matplotlib.org/stable/gallery/color/named_colors.html">https://matplotlib.org/stable/gallery/color/named_colors.html</a></p>
</dd>
<dt>neg_edge_color: <cite>str</cite> (default: ‘r’)</dt><dd><p>color for negative regulation arrow. A full list of accepted named colors can be found at:
<a class="reference external" href="https://matplotlib.org/stable/gallery/color/named_colors.html">https://matplotlib.org/stable/gallery/color/named_colors.html</a></p>
</dd>
<dt>arrowsize: <cite>float</cite> (default: 10)</dt><dd><p>size of interaction arrows</p>
</dd>
<dt>arrow_alpha: <cite>float</cite> (default: 0.75)</dt><dd><p>shading of interaction arrows in [0,1]</p>
</dd>
<dt>conn_style: <cite>str</cite> (default: ‘straight’)</dt><dd><p>style of interaction arrows. The admissible styles for networkx graphs can be found at:
<a class="reference external" href="https://networkx.org/documentation/stable/reference/generated/networkx.drawing.nx_pylab.draw_networkx_edges.html">https://networkx.org/documentation/stable/reference/generated/networkx.drawing.nx_pylab.draw_networkx_edges.html</a></p>
</dd>
<dt>colorbar: <cite>Bool</cite> (default: True)</dt><dd><p>if True, show colorbar (required if node_size=’expression’)</p>
</dd>
<dt>fontweight: <cite>str</cite> (default: ‘normal’)</dt><dd><p>style of text. Can select ‘bold’ for bold text.</p>
</dd>
</dl>
<p>None</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="splicejac.plot.grn_plots.visualize_network">
<span class="sig-prename descclassname"><span class="pre">splicejac.plot.grn_plots.</span></span><span class="sig-name descname"><span class="pre">visualize_network</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">adata</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cluster_name</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">genes</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cc_id</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">node_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'expression'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">edge_width</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'weight'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">font_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">plot_interactive</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">weight_quantile</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.5</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">node_color</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'centrality'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pos_style</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'spring'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">title</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">base_node_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">300</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">diff_node_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">600</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pos_edge_color</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'b'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">neg_edge_color</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'r'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">arrowsize</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">arrow_alpha</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.75</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">conn_style</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'straight'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">colorbar</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">fontweight</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'normal'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">showfig</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">savefig</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">format</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'pdf'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">figsize</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">(4,</span> <span class="pre">3)</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#splicejac.plot.grn_plots.visualize_network" title="Permalink to this definition"></a></dt>
<dd><p>Plot the gene regulatory network of a cluster</p>
<dl class="simple">
<dt>adata: <cite>~anndata.AnnData</cite></dt><dd><p>count matrix</p>
</dd>
<dt>cluster_name: <cite>str</cite></dt><dd><p>cell state</p>
</dd>
<dt>genes: <cite>list</cite> (default: <cite>None</cite>)</dt><dd><p>list of genes to include. If None, all genes are included</p>
</dd>
<dt>cc_id: <cite>int</cite> (default: 0)</dt><dd><p>connected component of the GRN to plot</p>
</dd>
<dt>node_size: <cite>int</cite> or <cite>str</cite> (default: 200)</dt><dd><p>size of nodes. If node_size=’expression’, the node size is scaled based on gene expression.
Otherwise, node_size can be a fixed integer</p>
</dd>
<dt>edge_width: <cite>float</cite> (default: 1)</dt><dd><p>width of connection arrows</p>
</dd>
<dt>fontsize: <cite>int</cite> (default: 10)</dt><dd><p>fontsize for node labels.</p>
</dd>
<dt>plot_interactive: <cite>Bool</cite> (default=True)</dt><dd><p>plot the GRN interactively in notebook of figure</p>
</dd>
<dt>weight_quantile: <cite>float</cite> (default=0.5)</dt><dd><p>threshold to filter weak interactions between 0 and 1</p>
</dd>
<dt>node_color: <cite>str</cite> (default: ‘centrality’)</dt><dd><p>color of nodes. If node_color=’centrality’, the color is based on the node centrality.
Otherwise, a matplotlib color name can be provided. A full list of accepted named colors can be found at:
<a class="reference external" href="https://matplotlib.org/stable/gallery/color/named_colors.html">https://matplotlib.org/stable/gallery/color/named_colors.html</a></p>
</dd>
<dt>pos_style: <cite>str</cite> (default: ‘spring’)</dt><dd><p>position of nodes. The options are ‘spring’ and ‘circle’. ‘spring’ uses the networkx spring position function,
whereas ‘circle’ arranges the nodes in a circle.</p>
</dd>
<dt>title: <cite>Bool</cite> (default=True)</dt><dd><p>if True, plot title</p>
</dd>
<dt>base_node_size: <cite>float</cite> (default: 300)</dt><dd><p>Minimum node size (used if node_size=’expression’)</p>
</dd>
<dt>diff_node_size: <cite>float</cite> (default: 600)</dt><dd><p>difference between minimum and maximal node size (used if node_size=’expression’)</p>
</dd>
<dt>pos_edge_color: <cite>str</cite> (default: ‘b’)</dt><dd><p>color for positive regulation arrow. A full list of accepted named colors can be found at:
<a class="reference external" href="https://matplotlib.org/stable/gallery/color/named_colors.html">https://matplotlib.org/stable/gallery/color/named_colors.html</a></p>
</dd>
<dt>neg_edge_color: <cite>str</cite> (default: ‘r’)</dt><dd><p>color for negative regulation arrow. A full list of accepted named colors can be found at:
<a class="reference external" href="https://matplotlib.org/stable/gallery/color/named_colors.html">https://matplotlib.org/stable/gallery/color/named_colors.html</a></p>
</dd>
<dt>arrowsize: <cite>float</cite> (default: 10)</dt><dd><p>size of interaction arrows</p>
</dd>
<dt>arrow_alpha: <cite>float</cite> (default: 0.75)</dt><dd><p>shading of interaction arrows in [0,1]</p>
</dd>
<dt>conn_style: <cite>str</cite> (default: ‘straight’)</dt><dd><p>style of interaction arrows. The admissible styles for networkx graphs can be found at:
<a class="reference external" href="https://networkx.org/documentation/stable/reference/generated/networkx.drawing.nx_pylab.draw_networkx_edges.html">https://networkx.org/documentation/stable/reference/generated/networkx.drawing.nx_pylab.draw_networkx_edges.html</a></p>
</dd>
<dt>colorbar: <cite>Bool</cite> (default: True)</dt><dd><p>if True, show colorbar (required if node_size=’expression’)</p>
</dd>
<dt>fontweight: <cite>str</cite> (default: ‘normal’)</dt><dd><p>style of text. Can select ‘bold’ for bold text.</p>
</dd>
<dt>showfig: <cite>Bool</cite> or <cite>None</cite> (default: <cite>None</cite>)</dt><dd><p>if True, show the figure</p>
</dd>
<dt>savefig: <cite>Bool</cite> or <cite>None</cite> (default: <cite>None</cite>)</dt><dd><p>if True, save the figure using the savefig path</p>
</dd>
<dt>format: <cite>str</cite> (default: ‘pdf’)</dt><dd><p>figure format</p>
</dd>
<dt>figsize: <cite>tuple</cite> (default: (5,4))</dt><dd><p>size of figure</p>
</dd>
</dl>
<p>None</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="splicejac.plot.grn_plots.diff_network">
<span class="sig-prename descclassname"><span class="pre">splicejac.plot.grn_plots.</span></span><span class="sig-name descname"><span class="pre">diff_network</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">adata</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cluster1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cluster2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">genes</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cc_id</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">node_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">500</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">edge_width</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'weight'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">font_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">weight_quantile</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.5</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pos_style</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'spring'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">base_node_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">300</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">diff_node_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">600</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pos_edge_color</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'b'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">neg_edge_color</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'r'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">arrowsize</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">arrow_alpha</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.75</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">conn_style</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'straight'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">colorbar</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">fontweight</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'normal'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">title</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">showfig</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">savefig</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">format</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'pdf'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">figsize</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">(3.5,</span> <span class="pre">3)</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#splicejac.plot.grn_plots.diff_network" title="Permalink to this definition"></a></dt>
<dd><p>Plot the differential network between two cell states</p>
<dl class="simple">
<dt>adata: <cite>~anndata.AnnData</cite></dt><dd><p>count matrix</p>
</dd>
<dt>cluster1: <cite>str</cite></dt><dd><p>first cell state</p>
</dd>
<dt>cluster2: <cite>str</cite></dt><dd><p>second cell state</p>
</dd>
<dt>genes: <cite>list</cite> (default: <cite>None</cite>)</dt><dd><p>list of genes to include. If None, all genes are included</p>
</dd>
<dt>cc_id: <cite>int</cite> (default: 0)</dt><dd><p>connected component of the GRN to plot</p>
</dd>
<dt>node_size: <cite>float</cite> (default=500)</dt><dd><p>size of nodes in the GRN</p>
</dd>
<dt>edge_width: <cite>float</cite> or <cite>str</cite> (default=’weight’)</dt><dd><p>width of GRN edges. If edge_width=’weight’, the edge width is proportional to the interaction strength.</p>
</dd>
<dt>fontsize: <cite>int</cite> (default: 10)</dt><dd><p>fontsize for figure.</p>
</dd>
<dt>weight_quantile: <cite>float</cite> (default: 0.5)</dt><dd><p>threshold to filter weak interactions between 0 and 1</p>
</dd>
<dt>pos_style: <cite>str</cite> (default: ‘spring’)</dt><dd><p>position of nodes. The options are ‘spring’ and ‘circle’. ‘spring’ uses the networkx spring position function,
whereas ‘circle’ arranges the nodes in a circle.</p>
</dd>
<dt>base_node_size: <cite>float</cite> (default: 300)</dt><dd><p>Minimum node size (used if node_size=’expression’)</p>
</dd>
<dt>diff_node_size: <cite>float</cite> (default: 600)</dt><dd><p>difference between minimum and maximal node size (used if node_size=’expression’)</p>
</dd>
<dt>pos_edge_color: <cite>str</cite> (default: ‘b’)</dt><dd><p>color for positive regulation arrow. A full list of accepted named colors can be found at:
<a class="reference external" href="https://matplotlib.org/stable/gallery/color/named_colors.html">https://matplotlib.org/stable/gallery/color/named_colors.html</a></p>
</dd>
<dt>neg_edge_color: <cite>str</cite> (default: ‘r’)</dt><dd><p>color for negative regulation arrow. A full list of accepted named colors can be found at:
<a class="reference external" href="https://matplotlib.org/stable/gallery/color/named_colors.html">https://matplotlib.org/stable/gallery/color/named_colors.html</a></p>
</dd>
<dt>arrowsize: <cite>float</cite> (default: 10)</dt><dd><p>size of interaction arrows</p>
</dd>
<dt>arrow_alpha: <cite>float</cite> (default: 0.75)</dt><dd><p>shading of interaction arrows in [0,1]</p>
</dd>
<dt>conn_style: <cite>str</cite> (default: ‘straight’)</dt><dd><p>style of interaction arrows. The admissible styles for networkx graphs can be found at:
<a class="reference external" href="https://networkx.org/documentation/stable/reference/generated/networkx.drawing.nx_pylab.draw_networkx_edges.html">https://networkx.org/documentation/stable/reference/generated/networkx.drawing.nx_pylab.draw_networkx_edges.html</a></p>
</dd>
<dt>colorbar: <cite>Bool</cite> (default: True)</dt><dd><p>if True, show colorbar (required if node_size=’expression’)</p>
</dd>
<dt>fontweight: <cite>str</cite> (default: ‘normal’)</dt><dd><p>style of text. Can select ‘bold’ for bold text.</p>
</dd>
<dt>title: <cite>Bool</cite> (default: True)</dt><dd><p>if True, plot title</p>
</dd>
<dt>showfig: <cite>Bool</cite> or <cite>None</cite> (default: <cite>None</cite>)</dt><dd><p>if True, show the figure</p>
</dd>
<dt>savefig: <cite>Bool</cite> or <cite>None</cite> (default: <cite>None</cite>)</dt><dd><p>if True, save the figure using the savefig path</p>
</dd>
<dt>format: <cite>str</cite> (default: ‘pdf’)</dt><dd><p>figure format</p>
</dd>
<dt>figsize: <cite>tuple</cite> (default: (3.5,3))</dt><dd><p>size of figure</p>
</dd>
</dl>
<p>None</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="splicejac.plot.grn_plots.diff_interactions">
<span class="sig-prename descclassname"><span class="pre">splicejac.plot.grn_plots.</span></span><span class="sig-name descname"><span class="pre">diff_interactions</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">adata</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cluster1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cluster2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">top_int</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">loc</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'best'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">title</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">fontsize</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">legend_font</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">legend_col</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">showfig</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">savefig</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">format</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'pdf'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">figsize</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">(4,</span> <span class="pre">5)</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#splicejac.plot.grn_plots.diff_interactions" title="Permalink to this definition"></a></dt>
<dd><p>Plot the top differential interactions between two cell states</p>
<dl class="simple">
<dt>adata: <cite>~anndata.AnnData</cite></dt><dd><p>count matrix</p>
</dd>
<dt>cluster1: <cite>str</cite></dt><dd><p>first cell state</p>
</dd>
<dt>cluster2: <cite>str</cite></dt><dd><p>second cell state</p>
</dd>
<dt>top_int: <cite>int</cite> (default=10)</dt><dd><p>number of top changed interactions to plot</p>
</dd>
<dt>loc: <cite>str</cite> (default=’best’)</dt><dd><p>location of legend. Details on legend location can be found at:
<a class="reference external" href="https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.legend.html">https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.legend.html</a></p>
</dd>
<dt>title: <cite>Bool</cite> (default: False)</dt><dd><p>if True, plot title</p>
</dd>
<dt>fontsize: <cite>int</cite> (default: 10)</dt><dd><p>fontsize for figure.</p>
</dd>
<dt>legend_font: <cite>int</cite> (default: 10)</dt><dd><p>fontsize of legend</p>
</dd>
<dt>legend_col: <cite>int</cite> (default: 1)</dt><dd><p>number of columns in legend</p>
</dd>
<dt>showfig: <cite>Bool</cite> or <cite>None</cite> (default: <cite>None</cite>)</dt><dd><p>if True, show the figure</p>
</dd>
<dt>savefig: <cite>Bool</cite> or <cite>None</cite> (default: <cite>None</cite>)</dt><dd><p>if True, save the figure using the savefig path</p>
</dd>
<dt>format: <cite>str</cite> (default: ‘pdf’)</dt><dd><p>figure format</p>
</dd>
<dt>figsize: <cite>tuple</cite> (default: (4,5))</dt><dd><p>size of figure</p>
</dd>
</dl>
<p>None</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="splicejac.plot.grn_plots.conserved_grn">
<span class="sig-prename descclassname"><span class="pre">splicejac.plot.grn_plots.</span></span><span class="sig-name descname"><span class="pre">conserved_grn</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">adata</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cluster1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cluster2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">genes</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cc_id</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">node_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">500</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">edge_width</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'weight'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">font_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">weight_quantile</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.5</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pos_style</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'spring'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">title</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">base_node_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">300</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">diff_node_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">600</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pos_edge_color</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'b'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">neg_edge_color</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'r'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">arrowsize</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">arrow_alpha</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.75</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">conn_style</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'straight'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">colorbar</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">fontweight</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'normal'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">showfig</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">savefig</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">format</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'pdf'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">figsize</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">(3.5,</span> <span class="pre">3)</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#splicejac.plot.grn_plots.conserved_grn" title="Permalink to this definition"></a></dt>
<dd><p>Plot the conserved network between two cell states</p>
<dl class="simple">
<dt>adata: <cite>~anndata.AnnData</cite></dt><dd><p>count matrix</p>
</dd>
<dt>cluster1: <cite>str</cite></dt><dd><p>first cell state</p>
</dd>
<dt>cluster2: <cite>str</cite></dt><dd><p>second cell state</p>
</dd>
<dt>genes: <cite>list</cite> or <cite>None</cite> (default: <cite>None</cite>)</dt><dd><p>list of genes to consider. If None, all genes are considered</p>
</dd>
<dt>cc_id: <cite>int</cite> (default: 0)</dt><dd><p>connected component of the GRN to plot</p>
</dd>
<dt>node_size: <cite>int</cite> (default: 500)</dt><dd><p>size of nodes in the GRN</p>
</dd>
<dt>edge_width: <cite>str</cite> or <cite>int</cite> (default: ‘weight’)</dt><dd><p>width of GRN edges. If edge_width=’weight’, the edge width is proportional to the interaction strength.</p>
</dd>
<dt>fontsize: <cite>int</cite> (default: 10)</dt><dd><p>fontsize for figure.</p>
</dd>
<dt>weight_quantile: <cite>float</cite> (default: 0.5)</dt><dd><p>threshold to filter weak interactions between 0 and 1</p>
</dd>
<dt>pos_style: <cite>str</cite> (default: ‘spring’)</dt><dd><p>position of nodes. The options are ‘spring’ and ‘circle’. ‘spring’ uses the networkx spring position function,
whereas ‘circle’ arranges the nodes in a circle.</p>
</dd>
<dt>title: <cite>Bool</cite> (default: True)</dt><dd><p>if True, plot title</p>
</dd>
<dt>base_node_size: <cite>float</cite> (default: 300)</dt><dd><p>Minimum node size (used if node_size=’expression’)</p>
</dd>
<dt>diff_node_size: <cite>float</cite> (default: 600)</dt><dd><p>difference between minimum and maximal node size (used if node_size=’expression’)</p>
</dd>
<dt>pos_edge_color: <cite>str</cite> (default: ‘b’)</dt><dd><p>color for positive regulation arrow. A full list of accepted named colors can be found at:
<a class="reference external" href="https://matplotlib.org/stable/gallery/color/named_colors.html">https://matplotlib.org/stable/gallery/color/named_colors.html</a></p>
</dd>
<dt>neg_edge_color: <cite>str</cite> (default: ‘r’)</dt><dd><p>color for negative regulation arrow. A full list of accepted named colors can be found at:
<a class="reference external" href="https://matplotlib.org/stable/gallery/color/named_colors.html">https://matplotlib.org/stable/gallery/color/named_colors.html</a></p>
</dd>
<dt>arrowsize: <cite>float</cite> (default: 10)</dt><dd><p>size of interaction arrows</p>
</dd>
<dt>arrow_alpha: <cite>float</cite> (default: 0.75)</dt><dd><p>shading of interaction arrows in [0,1]</p>
</dd>
<dt>conn_style: <cite>str</cite> (default: ‘straight’)</dt><dd><p>style of interaction arrows. The admissible styles for networkx graphs can be found at:
<a class="reference external" href="https://networkx.org/documentation/stable/reference/generated/networkx.drawing.nx_pylab.draw_networkx_edges.html">https://networkx.org/documentation/stable/reference/generated/networkx.drawing.nx_pylab.draw_networkx_edges.html</a></p>
</dd>
<dt>colorbar: <cite>Bool</cite> (default: True)</dt><dd><p>if True, show colorbar (required if node_size=’expression’)</p>
</dd>
<dt>fontweight: <cite>str</cite> (default: ‘normal’)</dt><dd><p>style of text. Can select ‘bold’ for bold text.</p>
</dd>
<dt>showfig: <cite>Bool</cite> or <cite>None</cite> (default: <cite>None</cite>)</dt><dd><p>if True, show the figure</p>
</dd>
<dt>savefig: <cite>Bool</cite> or <cite>None</cite> (default: <cite>None</cite>)</dt><dd><p>if True, save the figure using the savefig path</p>
</dd>
<dt>format: <cite>str</cite> (default: ‘pdf’)</dt><dd><p>figure format</p>
</dd>
<dt>figsize: <cite>tuple</cite> (default: (3.5,3))</dt><dd><p>size of figure</p>
</dd>
</dl>
<p>None</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="splicejac.plot.grn_plots.top_conserved_int">
<span class="sig-prename descclassname"><span class="pre">splicejac.plot.grn_plots.</span></span><span class="sig-name descname"><span class="pre">top_conserved_int</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">adata</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cluster1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cluster2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">top_int</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">title</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">fontsize</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">alpha</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.5</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">showfig</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">savefig</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">format</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'pdf'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">figsize</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">(4,</span> <span class="pre">5)</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#splicejac.plot.grn_plots.top_conserved_int" title="Permalink to this definition"></a></dt>
<dd><p>Plot the top conserved interactions between two cell states</p>
<dl class="simple">
<dt>adata: <cite>~anndata.AnnData</cite></dt><dd><p>count matrix</p>
</dd>
<dt>cluster1: <cite>str</cite></dt><dd><p>first cell state</p>
</dd>
<dt>cluster2: <cite>str</cite></dt><dd><p>second cell state</p>
</dd>
<dt>top_int: <cite>int</cite> (default=10)</dt><dd><p>number of top chnaged interactions to plot</p>
</dd>
<dt>title: <cite>Bool</cite> (default: False)</dt><dd><p>if True, plot title</p>
</dd>
<dt>fontsize: <cite>int</cite> (default: 10)</dt><dd><p>fontsize for figure.</p>
</dd>
<dt>alpha: <cite>float</cite> (default=0.5)</dt><dd><p>shading of bar plot</p>
</dd>
<dt>showfig: <cite>Bool</cite> or <cite>None</cite> (default: <cite>None</cite>)</dt><dd><p>if True, show the figure</p>
</dd>
<dt>savefig: <cite>Bool</cite> or <cite>None</cite> (default: <cite>None</cite>)</dt><dd><p>if True, save the figure using the savefig path</p>
</dd>
<dt>format: <cite>str</cite> (default: ‘pdf’)</dt><dd><p>figure format</p>
</dd>
<dt>figsize: <cite>tuple</cite> (default: (4,5))</dt><dd><p>size of figure</p>
</dd>
</dl>
<p>None</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="splicejac.plot.grn_plots.robust_mean">
<span class="sig-prename descclassname"><span class="pre">splicejac.plot.grn_plots.</span></span><span class="sig-name descname"><span class="pre">robust_mean</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">X</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#splicejac.plot.grn_plots.robust_mean" title="Permalink to this definition"></a></dt>
<dd><p>Calculate gene expression mean based on quantile selection</p>
<dl class="simple">
<dt>X: ~numpy.ndarray`</dt><dd><p>array of gene expression values</p>
</dd>
</dl>
<dl class="simple">
<dt>robust mean: ~numpy.ndarray`</dt><dd><p>array of mean gene expression</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="splicejac.plot.grn_plots.NormalizeData">
<span class="sig-prename descclassname"><span class="pre">splicejac.plot.grn_plots.</span></span><span class="sig-name descname"><span class="pre">NormalizeData</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">data</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#splicejac.plot.grn_plots.NormalizeData" title="Permalink to this definition"></a></dt>
<dd><p>normalize the gene expression data</p>
<dl class="simple">
<dt>data: ~numpy.ndarray`</dt><dd><p>data array</p>
</dd>
</dl>
<dl class="simple">
<dt>scaled: ~numpy.ndarray`</dt><dd><p>normalized data</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="splicejac.plot.grn_plots.circle_pos">
<span class="sig-prename descclassname"><span class="pre">splicejac.plot.grn_plots.</span></span><span class="sig-name descname"><span class="pre">circle_pos</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#splicejac.plot.grn_plots.circle_pos" title="Permalink to this definition"></a></dt>
<dd><p>Compute coordinates along a unit circle for GRN node positions</p>
<dl class="simple">
<dt>G: <cite>networkx object</cite></dt><dd><p>the networkx network</p>
</dd>
</dl>
<dl class="simple">
<dt>pos: ~numpy.ndarray`</dt><dd><p>array of positions</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="splicejac.plot.grn_plots.subset_jacobian">
<span class="sig-prename descclassname"><span class="pre">splicejac.plot.grn_plots.</span></span><span class="sig-name descname"><span class="pre">subset_jacobian</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">J</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">genes</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">genelist</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#splicejac.plot.grn_plots.subset_jacobian" title="Permalink to this definition"></a></dt>
<dd><p>select gene-gene interactions for genes in the genelist</p>
<dl class="simple">
<dt>J: <cite>~numpy.ndarray</cite></dt><dd><p>gene-gene interaction matrix</p>
</dd>
<dt>genes: <cite>list</cite></dt><dd><p>full list of genes</p>
</dd>
<dt>genelist: <cite>list</cite></dt><dd><p>list of selected genes</p>
</dd>
</dl>
<dl class="simple">
<dt>B_subset: <cite>~numpy.ndarray</cite></dt><dd><p>reduced gene-gene interaction matrix</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="splicejac.plot.grn_plots.core_GRN">
<span class="sig-prename descclassname"><span class="pre">splicejac.plot.grn_plots.</span></span><span class="sig-name descname"><span class="pre">core_GRN</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">adata</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cluster1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cluster2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">type_color</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">['orange',</span> <span class="pre">'plum',</span> <span class="pre">'yellowgreen']</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pos_edge_color</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'b'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">neg_edge_color</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'r'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">node_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">500</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">node_alpha</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.5</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">arrowsize</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">arrow_alpha</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.75</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">conn_style</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'arc3,</span> <span class="pre">rad=0.1'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">node_font</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">8</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">legend</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">legend_font</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">legend_ncol</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">legend_loc</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'lower</span> <span class="pre">center'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">axis</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">xlim</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">[-</span> <span class="pre">1.2,</span> <span class="pre">1.2]</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ylim</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">showfig</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">savefig</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">format</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'pdf'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">figsize</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">(3,</span> <span class="pre">3)</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#splicejac.plot.grn_plots.core_GRN" title="Permalink to this definition"></a></dt>
<dd><p>Plot a reduced GRN including the top DEG of the starting cluster and the top transition genes</p>
<dl class="simple">
<dt>adata: <cite>~anndata.AnnData</cite></dt><dd><p>count matrix</p>
</dd>
<dt>cluster1: <cite>str</cite></dt><dd><p>first cell state</p>
</dd>
<dt>cluster2: <cite>str</cite></dt><dd><p>second cell state</p>
</dd>
<dt>type_color: <cite>list</cite> (default=[‘orange’, ‘plum’, ‘yellowgreen’])</dt><dd><p>colors for nodes that are DEG, transition genes and both. A list of accepted colors can be found at:
<a class="reference external" href="https://matplotlib.org/stable/gallery/color/named_colors.html">https://matplotlib.org/stable/gallery/color/named_colors.html</a></p>
</dd>
<dt>pos_edge_color: <cite>str</cite> (default: ‘b’)</dt><dd><p>color for positive regulation arrow. A full list of accepted named colors can be found at:
<a class="reference external" href="https://matplotlib.org/stable/gallery/color/named_colors.html">https://matplotlib.org/stable/gallery/color/named_colors.html</a></p>
</dd>
<dt>neg_edge_color: <cite>str</cite> (default: ‘r’)</dt><dd><p>color for negative regulation arrow. A full list of accepted named colors can be found at:
<a class="reference external" href="https://matplotlib.org/stable/gallery/color/named_colors.html">https://matplotlib.org/stable/gallery/color/named_colors.html</a></p>
</dd>
<dt>node_size: ‘int’ (default: 500)</dt><dd><p>size of nodes</p>
</dd>
<dt>node_alpha: <cite>float</cite> (default: 0.5)</dt><dd><p>shading of nodes</p>
</dd>
<dt>arrowsize: <cite>float</cite> (default: 10)</dt><dd><p>size of interaction arrows</p>
</dd>
<dt>arrow_alpha: <cite>float</cite> (default: 0.75)</dt><dd><p>shading of interaction arrows in [0,1]</p>
</dd>
<dt>conn_style: <cite>str</cite> (default: ‘arc3, rad=0.1’)</dt><dd><p>style of interaction arrows. The admissible styles for networkx graphs can be found at:
<a class="reference external" href="https://networkx.org/documentation/stable/reference/generated/networkx.drawing.nx_pylab.draw_networkx_edges.html">https://networkx.org/documentation/stable/reference/generated/networkx.drawing.nx_pylab.draw_networkx_edges.html</a></p>
</dd>
<dt>node_font: <cite>int</cite> (default: 8)</dt><dd><p>fontsize for node labels.</p>
</dd>
<dt>legend: <cite>Bool</cite> (default=True)</dt><dd><p>if True, include legend</p>
</dd>
<dt>legend_font: <cite>int</cite> (default=10)</dt><dd><p>font size for legend</p>
</dd>
<dt>legend_ncol: <cite>int</cite> (default=2)</dt><dd><p>number of columns in legend</p>
</dd>
<dt>legend_loc: <cite>str</cite> (default: ‘lower center’)</dt><dd><p>legend location.  Details on legend location can be found at:
<a class="reference external" href="https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.legend.html">https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.legend.html</a></p>
</dd>
<dt>axis: <cite>Bool</cite> (default=False)</dt><dd><p>if True, plot axes</p>
</dd>
<dt>xlim: <cite>list</cite> (default=[-1.2, 1.2])</dt><dd><p>inteval on x-axis</p>
</dd>
<dt>ylim: <cite>list</cite> or <cite>None</cite> (default=None)</dt><dd><p>inteval on y-axis</p>
</dd>
<dt>showfig: <cite>Bool</cite> or <cite>None</cite> (default: <cite>None</cite>)</dt><dd><p>if True, show the figure</p>
</dd>
<dt>savefig: <cite>Bool</cite> or <cite>None</cite> (default: <cite>None</cite>)</dt><dd><p>if True, save the figure using the savefig path</p>
</dd>
<dt>format: <cite>str</cite> (default: ‘pdf’)</dt><dd><p>figure format</p>
</dd>
<dt>figsize: <cite>tuple</cite> (default: (3,3))</dt><dd><p>size of figure</p>
</dd>
</dl>
<p>None</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="splicejac.plot.grn_plots.bif_GRN">
<span class="sig-prename descclassname"><span class="pre">splicejac.plot.grn_plots.</span></span><span class="sig-name descname"><span class="pre">bif_GRN</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">adata</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">start</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">end</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pos_edge_color</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'b'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">neg_edge_color</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'r'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">node_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">750</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">node_alpha</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.5</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">arrowsize</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">arrow_alpha</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.75</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">arrowstyle</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'arc3,</span> <span class="pre">rad=0.1'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">node_font</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">legend</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">legend_font</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">legend_ncol</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">3</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">legend_loc</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'lower</span> <span class="pre">center'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">axis</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">xlim</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">[-</span> <span class="pre">1.2,</span> <span class="pre">1.2]</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ylim</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">showfig</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">savefig</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">format</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'pdf'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">figsize</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">(6,</span> <span class="pre">6)</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#splicejac.plot.grn_plots.bif_GRN" title="Permalink to this definition"></a></dt>
<dd><p>Plot the reduced GRN of transition genes involved in different cell state transitions</p>
<dl class="simple">
<dt>adata: <cite>~anndata.AnnData</cite></dt><dd><p>count matrix</p>
</dd>
<dt>start: <cite>str</cite></dt><dd><p>starting cell state</p>
</dd>
<dt>end: <cite>list</cite></dt><dd><p>list of ending cell states</p>
</dd>
<dt>pos_edge_color: <cite>str</cite> (default: ‘b’)</dt><dd><p>color for positive regulation arrow. A full list of accepted named colors can be found at:
<a class="reference external" href="https://matplotlib.org/stable/gallery/color/named_colors.html">https://matplotlib.org/stable/gallery/color/named_colors.html</a></p>
</dd>
<dt>neg_edge_color: <cite>str</cite> (default: ‘r’)</dt><dd><p>color for negative regulation arrow. A full list of accepted named colors can be found at:
<a class="reference external" href="https://matplotlib.org/stable/gallery/color/named_colors.html">https://matplotlib.org/stable/gallery/color/named_colors.html</a></p>
</dd>
<dt>node_size: <cite>int</cite> (default: 500)</dt><dd><p>size of nodes</p>
</dd>
<dt>node_alpha: <cite>float</cite> (default: 0.5)</dt><dd><p>shading of nodes</p>
</dd>
<dt>arrowsize: <cite>float</cite> (default: 10)</dt><dd><p>size of interaction arrows</p>
</dd>
<dt>arrow_alpha: <cite>float</cite> (default: 0.75)</dt><dd><p>shading of interaction arrows in [0,1]</p>
</dd>
<dt>conn_style: <cite>str</cite> (default: ‘arc3, rad=0.1’)</dt><dd><p>style of interaction arrows. The admissible styles for networkx graphs can be found at:
<a class="reference external" href="https://networkx.org/documentation/stable/reference/generated/networkx.drawing.nx_pylab.draw_networkx_edges.html">https://networkx.org/documentation/stable/reference/generated/networkx.drawing.nx_pylab.draw_networkx_edges.html</a></p>
</dd>
<dt>node_font: <cite>int</cite> (default=8)</dt><dd><p>font size of node labels</p>
</dd>
<dt>legend: <cite>Bool</cite> (default=True)</dt><dd><p>if True, include legend</p>
</dd>
<dt>legend_font: <cite>int</cite> (default=10)</dt><dd><p>font size for legend</p>
</dd>
<dt>legend_ncol: <cite>int</cite> (default=2)</dt><dd><p>number of columns in legend</p>
</dd>
<dt>legend_loc: <cite>str</cite> (default=’lower center’)</dt><dd><p>legend location.  Details on legend location can be found at:
<a class="reference external" href="https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.legend.html">https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.legend.html</a></p>
</dd>
<dt>axis: <cite>Bool</cite> (default=False)</dt><dd><p>if True, plot axes</p>
</dd>
<dt>xlim: <cite>list</cite> (default=[-1.2, 1.2])</dt><dd><p>inteval on x-axis</p>
</dd>
<dt>ylim: <cite>list</cite> or <cite>None</cite> (default=None)</dt><dd><p>inteval on y-axis</p>
</dd>
<dt>showfig: <cite>Bool</cite> or <cite>None</cite> (default: <cite>None</cite>)</dt><dd><p>if True, show the figure</p>
</dd>
<dt>savefig: <cite>Bool</cite> or <cite>None</cite> (default: <cite>None</cite>)</dt><dd><p>if True, save the figure using the savefig path</p>
</dd>
<dt>format: <cite>str</cite> (default: ‘pdf’)</dt><dd><p>figure format</p>
</dd>
<dt>figsize: <cite>tuple</cite> (default: (6,6))</dt><dd><p>size of figure</p>
</dd>
</dl>
<p>None</p>
</dd></dl>

</section>
</section>
</section>


           </div>
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2022, Federico Bocci.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>