<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>spliceJAC: Identify transition driver genes and cell state specific regulatory interactions from single-cell transcriptome data &mdash; spliceJAC 0.1 documentation</title>
      <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
        <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Usage" href="usage.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="#" class="icon icon-home"> spliceJAC
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="usage.html">Usage</a></li>
<li class="toctree-l1"><a class="reference internal" href="notebooks/GRN%20Inference.html">GRN Inference</a></li>
<li class="toctree-l1"><a class="reference internal" href="notebooks/Transitions.html">Transitions</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="#">spliceJAC</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="#" class="icon icon-home"></a> &raquo;</li>
      <li>spliceJAC: Identify transition driver genes and cell state specific regulatory interactions from single-cell transcriptome data</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/index.md.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  
<style>
/* CSS overrides for sphinx_rtd_theme */

/* 24px margin */
.nbinput.nblast.container,
.nboutput.nblast.container {
    margin-bottom: 19px;  /* padding has already 5px */
}

/* ... except between code cells! */
.nblast.container + .nbinput.container {
    margin-top: -19px;
}

.admonition > p:before {
    margin-right: 4px;  /* make room for the exclamation icon */
}

/* Fix math alignment, see https://github.com/rtfd/sphinx_rtd_theme/pull/686 */
.math {
    text-align: unset;
}
</style>
<section id="splicejac-identify-transition-driver-genes-and-cell-state-specific-regulatory-interactions-from-single-cell-transcriptome-data">
<h1>spliceJAC: Identify transition driver genes and cell state specific regulatory interactions from single-cell transcriptome data<a class="headerlink" href="#splicejac-identify-transition-driver-genes-and-cell-state-specific-regulatory-interactions-from-single-cell-transcriptome-data" title="Permalink to this headline"></a></h1>
<p>spliceJAC is a python-based toolkit to reconstruct cell state-specific gene regulatory networks (GRN) and predict transition driver genes leading to cell differentiation.</p>
<section id="overview-of-splicejac">
<h2>Overview of spliceJAC<a class="headerlink" href="#overview-of-splicejac" title="Permalink to this headline"></a></h2>
<p>spliceJAC requires the spliced (S) and unspliced (U) count matrices from single cell RNA-sequencing (scRNA-seq) experiment as well as cell annotations that are used to identify the cell states in the dataset. Starting from this information, spliceJAC builds a local mRNA splicing model that provides information about cell state specific gene regulatory interactions and driver genes that induce cell state transitions.</p>
<p><img alt="spliceJAC schematic" src="_images/spliceJAC.png" /></p>
</section>
<section id="applications">
<h2>Applications<a class="headerlink" href="#applications" title="Permalink to this headline"></a></h2>
<p>Application of spliceJAC include:</p>
<ul class="simple">
<li><p>Investigate the context-specific signaling role of genes in different cell states.</p></li>
<li><p>Analyze multi-stable systems where several cell states can coexist.</p></li>
<li><p>Distinguish the transition driver genes leading to distinct differentiation paths stemming from a common initial state.</p></li>
</ul>
</section>
<section id="use-and-installation">
<h2>Use and installation<a class="headerlink" href="#use-and-installation" title="Permalink to this headline"></a></h2>
<section id="install-the-package-from-pypl">
<h3>Install the package from PyPl<a class="headerlink" href="#install-the-package-from-pypl" title="Permalink to this headline"></a></h3>
<p>spliceJAC can be installed using pip:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="go">pip install -U splicejac</span>
</pre></div>
</div>
</section>
<section id="working-within-the-splicejac-repository">
<h3>Working within the spliceJAC repository<a class="headerlink" href="#working-within-the-splicejac-repository" title="Permalink to this headline"></a></h3>
<p>Alternatively, you can zip download the spliceJAC repository from the green ‘Code’ bottom at the top right corner of this page and run your custom code and notebooks from within the repository.</p>
</section>
<section id="setting-up-a-virtual-environment">
<h3>Setting up a virtual environment<a class="headerlink" href="#setting-up-a-virtual-environment" title="Permalink to this headline"></a></h3>
<p>Whether you decide to install the package or to work within the spliceJAC Repo, we suggest to work within a virtual environment to avoid conflict and ensure that all dependencies are updated to the required version. Guidelines to create a virtual environment in Python can be found <a class="reference external" href="https://docs.python.org/3/library/venv.html">here</a>.</p>
<p>spliceJAC requires the installation of dependencies including <a class="reference external" href="https://numpy.org">Numpy</a>, <a class="reference external" href="https://matplotlib.org">Matplotlib</a>, <a class="reference external" href="https://pandas.pydata.org">Pandas</a>, <a class="reference external" href="https://pandas.pydata.org">Scanpy</a> and <a class="reference external" href="https://scvelo.readthedocs.io">scVelo</a>. The full list of required packages can be found in the requirements.txt file within this folder. Once your virtual environment is set up, you can install all required dependencies by running:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="go">pip install -r requirements.txt</span>
</pre></div>
</div>
</section>
</section>
<section id="getting-started">
<h2>Getting started<a class="headerlink" href="#getting-started" title="Permalink to this headline"></a></h2>
<p>This repository includes two in-depth notebook tutorials to demonstrate spliceJAC’s applications in inferring cell state specific gene regulatory networks and analyze cell state transitions:</p>
<ul class="simple">
<li><p>GRN inference</p></li>
<li><p>Transitions</p></li>
</ul>
</section>
<section id="references">
<h2>References<a class="headerlink" href="#references" title="Permalink to this headline"></a></h2>
<p>Federico Bocci, Peijie Zhou, Qing Nie, spliceJAC: Identify transition driver genes and cell state specific regulatory interactions from single-cell transcriptome data, Preprint (2022).</p>
<p>The detailed testing and benchmarking of spliceJAC is publicly available <a class="reference external" href="https://github.com/cliffzhou92/jacobian-inference-benchmarking">here</a>.</p>
</section>
<section id="further-reading">
<h2>Further reading<a class="headerlink" href="#further-reading" title="Permalink to this headline"></a></h2>
<p>Bergen et al. (2020), Generalizing RNA velocity to transient cell states through dynamical modeling, <a class="reference external" href="https://www.nature.com/articles/s41587-020-0591-3">Nature Biotechnology</a>.</p>
<p>Pratapa et al. (2020), Benchmarking algorithms for gene regulatory network inference from single cell transcriptomic data, <a class="reference external" href="https://www.nature.com/articles/s41592-019-0690-6">Nature Methods</a>.</p>
<p>Wolf et al. (2018), SCANPY: large-scale single-cell gene expression data analysis, <a class="reference external" href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-017-1382-0">Genome Biology</a>.</p>
</section>
<section id="contact-and-support">
<h2>Contact and support<a class="headerlink" href="#contact-and-support" title="Permalink to this headline"></a></h2>
<p>Questions or suggestions are welcomed and highly appreciated. You can contact us by opening an issue or via email at fbocci&#64;uci.edu.</p>
<p>For further info, check out <a class="reference internal" href="usage.html"><span class="doc">Usage</span></a>.
Read installation instructions in <a class="reference internal" href="usage.html#installation"><span class="std std-ref">Installation</span></a></p>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>This library is under heavy developemnt.</p>
</div>
<div class="toctree-wrapper compound">
<p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="usage.html">Usage</a><ul>
<li class="toctree-l2"><a class="reference internal" href="usage.html#installation">Installation</a></li>
<li class="toctree-l2"><a class="reference internal" href="usage.html#functions">Functions</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="notebooks/GRN%20Inference.html">GRN Inference</a><ul>
<li class="toctree-l2"><a class="reference internal" href="notebooks/GRN%20Inference.html#Data-loading-and-preprocessing">Data loading and preprocessing</a></li>
<li class="toctree-l2"><a class="reference internal" href="notebooks/GRN%20Inference.html#Running-spliceJAC">Running spliceJAC</a></li>
<li class="toctree-l2"><a class="reference internal" href="notebooks/GRN%20Inference.html#Overview-of-state-specific-GRN-and-stability-considerations">Overview of state-specific GRN and stability considerations</a></li>
<li class="toctree-l2"><a class="reference internal" href="notebooks/GRN%20Inference.html#Visualize-the-GRNs-and-gene-signaling-roles">Visualize the GRNs and gene signaling roles</a></li>
<li class="toctree-l2"><a class="reference internal" href="notebooks/GRN%20Inference.html#Quantify-signaling-role-variation-across-cell-states">Quantify signaling role variation across cell states</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="notebooks/Transitions.html">Transitions</a><ul>
<li class="toctree-l2"><a class="reference internal" href="notebooks/Transitions.html#Preparing-the-data-for-spliceJAC-analysis">Preparing the data for spliceJAC analysis</a></li>
<li class="toctree-l2"><a class="reference internal" href="notebooks/Transitions.html#Analysis-of-signaling-change-during-transitions">Analysis of signaling change during transitions</a></li>
<li class="toctree-l2"><a class="reference internal" href="notebooks/Transitions.html#Identify-and-plot-the-top-transition-genes">Identify and plot the top transition genes</a></li>
<li class="toctree-l2"><a class="reference internal" href="notebooks/Transitions.html#Identify-the-differential-and-conserved-interactions-during-cell-state-transitions">Identify the differential and conserved interactions during cell state transitions</a></li>
<li class="toctree-l2"><a class="reference internal" href="notebooks/Transitions.html#Compare-transition-genes-in-bifurcating-lineages">Compare transition genes in bifurcating lineages</a></li>
</ul>
</li>
</ul>
</div>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="usage.html" class="btn btn-neutral float-right" title="Usage" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2022, Federico Bocci.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>