
<!DOCTYPE html>

<html lang="en">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

    <title>Usage &#8212; spliceJAC 0.1 documentation</title>
    <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="_static/alabaster.css" />
    <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/_sphinx_javascript_frameworks_compat.js"></script>
    <script src="_static/doctools.js"></script>
    <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="GRN Inference" href="notebooks/GRN%20Inference.html" />
    <link rel="prev" title="spliceJAC: Identify transition driver genes and cell state specific regulatory interactions from single-cell transcriptome data" href="index.html" />
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <section id="usage">
<h1>Usage<a class="headerlink" href="#usage" title="Permalink to this heading">¶</a></h1>
<section id="installation">
<h2>Installation<a class="headerlink" href="#installation" title="Permalink to this heading">¶</a></h2>
<p>spliceJAC can be installed using pip:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="gp gp-VirtualEnv">(.venv)</span> <span class="gp">$ </span>pip install -U splicejac
</pre></div>
</div>
</section>
<section id="functions">
<h2>Functions<a class="headerlink" href="#functions" title="Permalink to this heading">¶</a></h2>
<dl class="py function">
<dt class="sig sig-object py" id="splicejac.tl.estimate_jacobian">
<span class="sig-prename descclassname"><span class="pre">splicejac.tl.</span></span><span class="sig-name descname"><span class="pre">estimate_jacobian</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">adata</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">first_moment</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">method</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'Ridge'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">alpha</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">beta</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1.0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">rescale</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">nsim</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">frac</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.9</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">filter_and_norm</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">min_shared_counts</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">20</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">n_top_genes</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">20</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">eps</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.9</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">seed</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">100</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#splicejac.tl.estimate_jacobian" title="Permalink to this definition">¶</a></dt>
<dd><p>Run cluster-wise Jacobian inference</p>
<dl class="simple">
<dt>adata: <cite>~anndata.AnnData</cite></dt><dd><p>count matrix</p>
</dd>
<dt>first_moment: <cite>Bool</cite> (default: True)</dt><dd><p>if True, use first moment of U and S to run regression</p>
</dd>
<dt>method: <cite>str</cite> (default: Ridge)</dt><dd><p>regression method, choose between Linear, Ridge or Lasso</p>
</dd>
<dt>alpha: <cite>float</cite> (default: 1)</dt><dd><p>regularization coefficient for Ridge and Lasso</p>
</dd>
<dt>beta: <cite>float</cite> (default: 1)</dt><dd><p>mRNA splicing rate constant</p>
</dd>
<dt>rescale: <cite>Bool</cite> (default: True)</dt><dd><p>if True, center counts on zero (default= True). rescale=True enforces fit_int=False</p>
</dd>
<dt>nsim: <cite>int</cite> (default: 10)</dt><dd><p>number of independent regressions per cluster</p>
</dd>
<dt>frac: <cite>float</cite> (default: 0.9)</dt><dd><p>fraction of cells to randomly select (bound in [0,1])</p>
</dd>
<dt>filter_and_norm: <cite>Bool</cite> (default: True)</dt><dd><p>if True, apply scvelo filter_and_normalize function to the count matrix</p>
</dd>
<dt>min_shared_count: <cite>int</cite> (default: 20)</dt><dd><p>minimum number of shared count for the scvelo filter_and_normalize function</p>
</dd>
<dt>n_top_genes: <cite>int</cite> (default: 20)</dt><dd><p>number of top genes for the scvelo filter_and_normalize function</p>
</dd>
<dt>eps: <cite>float</cite> (default= 0.9)</dt><dd><p>fraction of weakest Jacobian elements that are set to zero (by absolute value)</p>
</dd>
<dt>seed: <cite>int</cite> (default: 100)</dt><dd><p>seed for numpy random number generator (for reproducibility)</p>
</dd>
</dl>
<p>None</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="splicejac.tl.trans_from_PAGA">
<span class="sig-prename descclassname"><span class="pre">splicejac.tl.</span></span><span class="sig-name descname"><span class="pre">trans_from_PAGA</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">adata</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dir_method</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'top_eig'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">eig_number</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">5</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">top_DEG</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">5</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">top_TG</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">5</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">first_moment</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#splicejac.tl.trans_from_PAGA" title="Permalink to this definition">¶</a></dt>
<dd><p>Compute the gene instability scores for all transitions identified with PAGA
PAGA transitions must be stored as a dataframe in adata.uns[‘PAGA_paths’]
results are stored in adata.uns[‘transitions’]</p>
<dl class="simple">
<dt>adata: <cite>~anndata.AnnData</cite></dt><dd><p>count matrix</p>
</dd>
<dt>dir_method: <cite>str</cite> (default: ‘top_eig’)</dt><dd><p>method to select the unstable directions, choose between ‘top_eig’ and ‘positive’.
‘top_eig’ uses the largest eigenvalues irrespective of sign; ‘positive’ strictly uses positive eigenvalues</p>
</dd>
<dt>eig_number: <cite>int</cite> (default: 5)</dt><dd><p>number of largest eigenvalues to consider, required for dir_method=’top_eig’</p>
</dd>
<dt>top_DEG: <cite>int</cite> (default: 5)</dt><dd><p>number of top DEG to select</p>
</dd>
<dt>top_TG: <cite>int</cite> (default: 5)</dt><dd><p>number of top TG to select</p>
</dd>
<dt>first_moment: <cite>Bool</cite> (default: True)</dt><dd><p>if True, use first moments of unspliced/spliced counts</p>
</dd>
</dl>
<p>None</p>
</dd></dl>

</section>
</section>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="index.html">spliceJAC</a></h1>








<h3>Navigation</h3>
<p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="#">Usage</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#installation">Installation</a></li>
<li class="toctree-l2"><a class="reference internal" href="#functions">Functions</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="notebooks/GRN%20Inference.html">GRN Inference</a></li>
<li class="toctree-l1"><a class="reference internal" href="notebooks/Transitions.html">Transitions</a></li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="index.html">Documentation overview</a><ul>
      <li>Previous: <a href="index.html" title="previous chapter">spliceJAC: Identify transition driver genes and cell state specific regulatory interactions from single-cell transcriptome data</a></li>
      <li>Next: <a href="notebooks/GRN%20Inference.html" title="next chapter">GRN Inference</a></li>
  </ul></li>
</ul>
</div>
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false"/>
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>document.getElementById('searchbox').style.display = "block"</script>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2022, Federico Bocci.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 5.1.1</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.12</a>
      
      |
      <a href="_sources/usage.md.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>